src/components/TodoList.tsx

```diff
@@ -0,0 +1,295 @@
+            className={classNames(
+              'todo',
+              { completed: todo.completed },
+            )}
+            onDoubleClick={() => {
+              onDoubleClick(index);
+              setInputValue(todo.title);
+              setIsTodoEditing(true);
+              setTimeout(() => {
+                if (inputRef) {
+                  inputRef.current?.focus();
+                }
+              }, 0);
+            }}
+          >
+            <label className={classNames(
+              'todo__status-label',
+              { hidden: isTodoEditing && index === clickedIndex },
+            )}
+            >
+              <input
+                data-cy="TodoStatus"
+                type="checkbox"
+                className="todo__status"
+                defaultChecked
+                onClick={() => {
+                  setClickedIndex(index);
+                  handleToggleTodo(todo.id, todo.title);
+                }}
+              />
+            </label>
+
+            {isTodoEditing && index === clickedIndex ? (
+              <form
+                onSubmit={(event) => onEditSubmit(
+                  event,
+                  todo.id,
+                  todo.completed,
+                )}
+                className="todo__title-form"
+              >
+                <input
+                  type="text"
+                  value={inputValue}
+                  onChange={(e) => setInputValue(e.target.value)}
+                  ref={inputRef}
+                  onBlur={() => setIsTodoEditing(false)}
+                  onKeyDown={onKeyDownHandler}
+                  className="todo__input"
+                />
+              </form>
+            ) : (
+              <span data-cy="TodoTitle" className="todo__title">
+                {todo.title}
+              </span>
+            )}
+
+            <button
+              type="button"
+              className={classNames(
+                'todo__remove',
+                { hidden: isTodoEditing && index === clickedIndex },
+              )}
+              data-cy="TodoDeleteButton"
+              onClick={() => {
+                setClickedIndex(index);
+                deleteTodoHandler(todo.id);
+              }}
+            >
+              ×
+            </button>
+
+            {index === clickedIndex && (
+              <Loader
+                isActiveCondition={((!isNewTodoLoaded))}
+              />
+            )}
+
+            {index === clickedIndex && (
+              <Loader isActiveCondition={isTodoToggled} />
+            )}
+
+            {index === clickedIndex && (
+              <Loader isActiveCondition={!isTodoDeleted} />
+            )}
+
+            {index === clickedIndex && (
+              <Loader isActiveCondition={!isTodoEdited} />
+            )}
+
+            <Loader
+              isActiveCondition={(todo.completed && isCompletedTodosDeleting)}
+            />
+
+            <Loader isActiveCondition={areTodosToggling} />
```

Comment:
You can create a function for example `getLoaderConfig` and get props for the Loader by using this function. It doesn't look good now.